src/snowflake/connector/cursor.py

```diff
@@ -274,6 +275,10 @@ def __del__(self) -> None:  # pragma: no cover
     def description(self) -> List[ResultMetadata]:
         return self._description
 
+    @property
+    def query(self) -> Optional[str]:
+        return self._query
```

Comment:
I'm not sure if I agree with making this a property, what do you think about having it as a regular attribute?

Reply:
I followed the pattern of other underscore prefixed attribute and property pairs, but as far as I see, this Java-like pattern just adds boilerplate.

Would you also be in favor of removing this pattern elsewhere? It's used quite heavily throughout the project.

Reply:
It makes sense in some cases. I would be for changing existing ones, but let's do that in another PR.